### PR TITLE
Rebuild dotc if new *.scala files are found in `dotty/src`

### PR DESCRIPTION
**Pull request by DarkDimius**

### CODE REVIEW

**Comment by DarkDimius** [comment]
ping @VladUreche, I guess you could reuse this.

**Comment by DarkDimius** [comment]
@smarter please review.

**Comment by smarter** [comment]
LGTM.

**Comment by VladUreche** [comment]
Thanks @DarkDimius! I keep an `sbt ~package` running during development, and it's fast enough for the miniboxing plugin. But I agree for a larger code base, like dotty, you don't want to build each change.

File: bin/dotc [hunk 1]
```diff
@@ -57,11 +57,21 @@ function checkjar {
...
       echo "The required jar file was built successfully."
     fi
+  else
+    NEW_FILES="$(find "$DOTTY_ROOT/$3" -iname "*.scala" -newer "$1")"
+    if [ ! -z "$NEW_FILES" ]; 
+    then
+      echo "new files detected. rebuilding"
+      cd $DOTTY_ROOT
+      sbt $2
+      touch "$1"
```
Comment (VladUreche) [inline bin/dotc]: Why do you need to do this manually? At a first sight, sbt should update the jar file when rebuilding, right?

Comment (DarkDimius) [inline bin/dotc]: Sometimes sbt is smart enough to see that it does not need to rebuild the project, though timestamps changed. Eg if you make a change and revert it.

**Pull request merged**